### PR TITLE
Remove api-version validation

### PR DESCRIPTION
**Pull request by sarangan12**
@vishrutshah @veronicagg @salameer @dsgouda  Could you please review and approve?

### CODE REVIEW

File: src/modeler/AutoRest.Swagger/Validation/OperationParametersValidation.cs [hunk 1]
```diff
@@ -30,7 +29,7 @@ public class OperationParametersValidation : TypedRule<SwaggerParameter>
         /// <param name="paths"></param>
         /// <returns></returns>
         public override bool IsValid(SwaggerParameter Parameter) => 
-           (!string.IsNullOrEmpty(Parameter.Reference) ||Parameter?.Schema != null || !(Parameter?.Name?.ToLower().Equals(SubscriptionId) == true || Parameter?.Name?.ToLower().Equals(ApiVersion) == true));
```
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Validation/OperationParametersValidation.cs]: You'll have to update the validation message and regenerate the code

File: src/modeler/AutoRest.Swagger/Validation/OperationParametersValidation.cs [hunk 2]
```diff
@@ -11,7 +11,6 @@ namespace AutoRest.Swagger.Validation
     public class OperationParametersValidation : TypedRule<SwaggerParameter>
     {
         private const string SubscriptionId = "subscriptionid";
-        private const string ApiVersion = "api-version";
```
Comment (dsgouda) [inline src/modeler/AutoRest.Swagger/Validation/OperationParametersValidation.cs]: Just add a to-do saying we need to add api-version in the check some time in the future when we enable the single-swagger spec mode for validation.

**Comment by veronicagg** [comment]
@sarangan12 @dsgouda Why are we removing this? could you please provide some context in the PR on this?

**Comment by dsgouda** [comment]
When composite

File: src/modeler/AutoRest.Swagger/Validation/OperationParametersValidation.cs (hunk 1 above)
Comment (sarangan12) [inline src/modeler/AutoRest.Swagger/Validation/OperationParametersValidation.cs]: Updated the message. What do you mean by regenerate the code?

**Comment by dsgouda** [comment]
Just run ```gulp build``` and ```gulp regenerate``` and update diffs if any.

**Comment by sarangan12** [comment]
@veronicagg 

The basic idea of the rule is that the subscription id and api version should not be defined inline and must be defined globally. For subscription id, it is not an issue. But, for the api-version it is. Why?

Autorest core treats api-version as special case. It automatically picks the definition of the api-version and substitutes it in the servicedefinition. So, at the time of execution of the validation, we will not know whether the api-version is defined inline or globally. This creates a lot of noise. So, removing this

**Comment by sarangan12** [comment]
@dsgouda regenerate did not add anything new.

**Comment by dsgouda** [comment]
While we are at it, do you think you could resolve [this](https://github.com/Azure/autorest/issues/2128) too, should be a very minimal change

**Comment by sarangan12** [comment]
@dsgouda That issue is not for this rule. It is for the ServiceDefinitionParameters. I am taking it up as next task. Not merging with this

**Comment by veronicagg** [comment]
@sarangan12 thanks for the explanation! based on our conversation, let's open an issue to see how we could validate api-version being the same across a swagger file, since that's one issue we would be missing by removing this check. It's basically that rule should apply to each individual file and not a composite, which is related to what @dsgouda and I have discussed as a feature to add to our validation .(https://github.com/Azure/autorest/issues/1940)

**Comment by sarangan12** [comment]
@veronicagg Sent a note to @kirthik

**Pull request merged**